app_test.go

```diff
@@ -5,98 +5,212 @@
+		testData := GetTestDataDeleteCursor()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "POST" {
+		checkContentType(response, request)
+		testData := GetTestDataCreateCursor()
+		fmt.Fprint(response, testData.output)
 	}
+}
 
-	if rec, err := a.GetRecord(116); err != nil {
-		t.Error(err)
-	} else {
-		if rec.Id() != 116 {
-			t.Errorf("Unexpected Id: %d", rec.Id())
-		}
-		for _, f := range rec.Fields {
-			if files, ok := f.(FileField); ok {
-				if len(files) == 0 {
-					continue
-				}
-				fd, err := a.Download(files[0].FileKey)
-				if err != nil {
-					t.Error(err)
-				} else {
-					data, _ := ioutil.ReadAll(fd.Reader)
-					t.Logf("%s %d bytes", fd.ContentType, len(data))
-				}
-			}
-		}
+func handleResponseRecordComments(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "POST" {
+		checkContentType(response, request)
+		testData := GetTestDataAddRecordComment()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "DELETE" {
+		checkContentType(response, request)
+		testData := GetDataTestDeleteRecordComment()
+		fmt.Fprint(response, testData.output)
 	}
+}
 
-	if recs, err := a.GetRecords(nil, "limit 3 offset 3"); err != nil {
-		t.Error(err)
-	} else {
-		if len(recs) > 3 {
-			t.Error("Too many records")
-		}
+func handleResponseUploadFile(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "POST" {
+		testData := GetDataTestUploadFile()
+		fmt.Fprint(response, testData.output)
 	}
+}
 
-	if recs, err := a.GetAllRecords([]string{"レコード番号"}); err != nil {
-		t.Error(err)
-	} else {
-		t.Log(len(recs))
+func handleResponseGetRecord(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "GET" {
+		checkContentType(response, request)
+		testData := GetTestDataGetRecord()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "PUT" {
+		checkContentType(response, request)
+		testData := GetTestDataUpdateRecordByKey()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "POST" {
+		checkContentType(response, request)
+		testData := GetTestDataAddRecord()
+		fmt.Fprint(response, testData.output)
 	}
+
 }
-func TestAddRecord(t *testing.T) {
-	a := newApp(9004)
-	if len(a.Password) == 0 {
-		t.Skip()
+
+func handleResponseGetRecords(response http.ResponseWriter, request *http.Request) {
+	checkAuth(response, request)
+	if request.Method == "GET" {
+		checkContentType(response, request)
+		testData := GetTestDataGetRecords()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "DELETE" {
+		checkContentType(response, request)
+		testData := GetTestDataDeleteRecords()
+		fmt.Fprint(response, testData.output)
+	} else if request.Method == "POST" {
+		checkContentType(response, request)
+		testData := GetTestDataAddRecords()
+		fmt.Fprint(response, testData.output)
+	}
```

Comment:
If they are all used "checkContentType" please move it to out side for common using.